packages/ui/src/composites/wui-wallet-image/index.ts

```diff
@@ -0,0 +1,45 @@
+import { html, LitElement } from 'lit'
+import { customElement, property } from 'lit/decorators.js'
+import { globalStyles } from '../../utils/ThemeUtil'
+import '../../components/wui-image'
+import '../../components/wui-icon'
+import styles from './styles'
+import { walletPlaceholderSvg } from '../../assets/svg/wallet-placeholder'
+import type { Size } from '../../utils/TypesUtil'
+
+@customElement('wui-wallet-image')
+export class WuiWalletImage extends LitElement {
+  public static styles = [globalStyles, styles]
+
+  // -- state & properties ------------------------------------------- //
+  @property() public size: Exclude<Size, 'inherit' | 'xs' | 'xxs'> = 'md'
+
+  @property() public walletName = ''
+
+  @property() public src?: string
+
+  // -- render ------------------------------------------------------- //
+  public render() {
+    const sizeClass = `wui-size-${this.size}`
+
+    function isImageUrl(url: string): boolean {
```

Comment:
In my opinion, un-necessary complication, there are other formats we might use like avif etc.. Just checking for empty `this.src` will be enough here. We actually want to see something going wrong if we provide non image url by accident.